### DISCUSSION TITLE
Recent Updates!

### DISCUSSION THREAD

**Discussion by bkonkle** [bkonkle/rust-example-caster-api#24] [Announcements]
:wave: Hi! I've made a lot of changes to this project over the last week!
- I reworked [the Github actions](https://github.com/bkonkle/rust-example-caster-api/tree/main/.github/workflows) to use more community actions for installing the Rust toolchain and caching Cargo operations.
- I added tools like [pre-commit](https://github.com/bkonkle/rust-example-caster-api/blob/main/.pre-commit-config.yaml), [llvm-cov](https://github.com/bkonkle/rust-example-caster-api/blob/main/Makefile.toml#L53-L56), and [cargo-audit](https://github.com/bkonkle/rust-example-caster-api/blob/main/Makefile.toml#L97-L100) to the mix.
- I moved to [nextest](https://nexte.st/) for faster test runs that take better advantage of concurrency.
- I added an "integration" feature [that disables JWT token validation](https://github.com/bkonkle/rust-example-caster-api/blob/main/libs/auth/src/authenticate.rs#L94-L129), so I can send any encoded JWT token I want in the integration tests without having to call out to my OAuth server, allowing me to more easily fake different scenarios for testing.
- This allowed me to then move to integration tests that don't rely on IDs that come from my OAuth server, but [instead use randomly generated unique IDs](https://github.com/bkonkle/rust-example-caster-api/blob/main/apps/api/tests/test_user_integration.rs#L39) - eliminating the possibility of collisions during integration testing and removing the need for test user credentials in the secrets.
- This enabled fully-concurrent integration testing, so that I could take advantage of nextest and _dramatically_ boost the execution speed for my integrations.
- And finally, I moved from Warp to [Axum](https://github.com/tokio-rs/axum), allowing for [much simpler route definitions](https://github.com/bkonkle/rust-example-caster-api/blob/main/apps/api/src/lib.rs#L122-L133).

I'll definitely be working on some video content soon to highlight everything!

**Comment by Dindaleon** [comment]
Amazing stuff, keep it up! Looking forward to watch the video. Will it be posted on youtube?

**Reply by bkonkle** [reply]
Thank you! Yes, I'll post it to YouTube, probably next week around Friday. 👍
